selfservice/strategy/oidc/provider_spotify.go

```diff
@@ -0,0 +1,95 @@
+package oidc
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+
+	"golang.org/x/oauth2/spotify"
+
+	"github.com/pkg/errors"
+	"golang.org/x/oauth2"
+
+	"github.com/ory/x/stringslice"
+	"github.com/ory/x/stringsx"
+
+	spotifyapi "github.com/zmb3/spotify/v2"
+	spotifyauth "github.com/zmb3/spotify/v2/auth"
+
+	"github.com/ory/herodot"
+)
+
+type ProviderSpotify struct {
+	config *Configuration
+	public *url.URL
+}
+
+func NewProviderSpotify(
+	config *Configuration,
+	public *url.URL,
+) *ProviderSpotify {
+	return &ProviderSpotify{
+		config: config,
+		public: public,
+	}
+}
+
+func (g *ProviderSpotify) Config() *Configuration {
+	return g.config
+}
+
+func (g *ProviderSpotify) oauth2() *oauth2.Config {
+	return &oauth2.Config{
+		ClientID:     g.config.ClientID,
+		ClientSecret: g.config.ClientSecret,
+		Endpoint:     spotify.Endpoint,
+		Scopes:       g.config.Scope,
+		RedirectURL:  g.config.Redir(g.public),
+	}
+}
+
+func (g *ProviderSpotify) OAuth2(ctx context.Context) (*oauth2.Config, error) {
+	return g.oauth2(), nil
+}
+
+func (g *ProviderSpotify) AuthCodeURLOptions(r ider) []oauth2.AuthCodeOption {
+	return []oauth2.AuthCodeOption{}
+}
+
+func (g *ProviderSpotify) Claims(ctx context.Context, exchange *oauth2.Token) (*Claims, error) {
+	grantedScopes := stringsx.Splitx(fmt.Sprintf("%s", exchange.Extra("scope")), " ")
+	for _, check := range g.Config().Scope {
+		if !stringslice.Has(grantedScopes, check) {
+			return nil, errors.WithStack(ErrScopeMissing)
+		}
+	}
+
+	auth := spotifyauth.New(
+		spotifyauth.WithRedirectURL(g.config.Redir(g.public)),
+		spotifyauth.WithScopes(spotifyauth.ScopeUserReadPrivate))
+
+	client := spotifyapi.New(auth.Client(ctx, exchange))
+
+	user, err := client.CurrentUser(ctx)
+	if err != nil {
+		return nil, errors.WithStack(herodot.ErrInternalServerError.WithReasonf("%s", err))
+	}
+
+	var userPicture string
+	if len(user.Images) > 0 {
+		userPicture = user.Images[0].URL
+	}
+
+	claims := &Claims{
+		Subject:   user.ID,
+		Issuer:    spotify.Endpoint.TokenURL,
+		Name:      user.DisplayName,
+		Nickname:  user.DisplayName,
+		Email:     user.Email,
+		Picture:   userPicture,
+		Profile:   user.ExternalURLs["spotify"],
+		Birthdate: user.Birthdate,
```

Comment:
Hey there! I'm looking to make a contribution here in order to get Country from the Spotify API. Would it be simple as adding `Country: user.country` into the Claims struct here?

My apologies - I'm not familiar with the providers or Go. But I'd love to make my first very simple contribution to Kratos.

Should this change be accepted, what is the average turn-around on a release for a non-breaking change like this making it to Ory Cloud? This is a requirement for a system I am building.

Reply:
> Would it be simple as adding Country: user.country into the Claims struct here?

Not sure, but I suggest you try it with a local Kratos instance and your patch. Make sure to check the spotify docs on the exact key and scopes needed.

> what is the average turn-around on a release for a non-breaking change like this making it to Ory Cloud?

Days, but we can prioritize this when requested.

Reply:
> The country of the user, as set in the user's account profile. An [ISO 3166-1 alpha-2 country code](http://en.wikipedia.org/wiki/ISO_3166-1_alpha-2). This field is only available when the current user has granted access to the [user-read-private](https://developer.spotify.com/documentation/general/guides/authorization-guide/#list-of-scopes) scope.

from https://developer.spotify.com/documentation/web-api/reference/#/operations/get-current-users-profile